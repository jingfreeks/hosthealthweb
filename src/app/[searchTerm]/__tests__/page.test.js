import { render, baseRender } from "@testing-library/react";
import Page from "../page";

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () =>
      Promise.resolve({
        24884283: {
          pageid: 24884283,
          ns: 0,
          title: "TeST Gliders",
          index: 3,
          thumbnail: {
            source:
              "https://upload.wikimedia.org/wikipedia/commons/thumb/d/d7/104_b_aircraft_0.jpg/50px-104_b_aircraft_0.jpg",
            width: 50,
            height: 38,
          },
          pageimage: "104_b_aircraft_0.jpg",
          extract:
            "TeST Gliders was a manufacturer of ultralight sailplanes and motorgliders based in Brno, Czech Republic...",
        },
      }),
  })
);

describe("Search Term Page", () => {
  const props = {
    params: {
      searchTerm: "",
    },
  };
  it("renders Search Term Page to match snaptshots", async () => {
    // const { container } = render(
    //   await Page({ params: { searchTerm: "test" } })
    // );
    const tree = render(await Page({ params: { searchTerm: "test" } }));
    const head = tree.baseElement.parentElement?.firstChild;
    expect(head).toMatchSnapshot();
  });
});
