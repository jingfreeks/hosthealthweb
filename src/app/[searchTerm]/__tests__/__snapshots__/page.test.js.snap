// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Term Page renders Search Term Page to match snaptshots 1`] = `
<div>
  <main
    class="bg-slate-200 mx-auto max-w-lg py-1 min-h-screen"
  >
    <article
      class="m-3 max-w-lg"
    >
      <div
        class="flex flex-row gap-4"
      >
        <div
          class="flex flex-col justify-center"
        >
          <img
            alt="Intelligence quotient"
            height="50"
            loading="lazy"
            src="https://upload.wikimedia.org/wikipedia/commons/thumb/e/ec/Raven_Matrix.svg/50px-Raven_Matrix.svg.png"
            width="50"
          />
        </div>
        <div
          class="flex flex-col justify-center"
        >
          <h2>
            <a
              class="text-xl font-bold underline"
              href="https://en.wikipedia.org/?curid=14892"
              target="_blank"
            >
              Intelligence quotient
            </a>
          </h2>
          <p>
            An intelligence quotient (IQ) is a total score derived from a set of standardised tests or subtests designed...
          </p>
        </div>
      </div>
    </article>
    <article
      class="m-3 max-w-lg"
    >
      <div
        class="flex flex-row gap-4"
      >
        <div
          class="flex flex-col justify-center"
        >
          <img
            alt="Test cricket"
            height="33"
            loading="lazy"
            src="https://upload.wikimedia.org/wikipedia/commons/thumb/1/13/England_vs_South_Africa.jpg/50px-England_vs_South_Africa.jpg"
            width="50"
          />
        </div>
        <div
          class="flex flex-col justify-center"
        >
          <h2>
            <a
              class="text-xl font-bold underline"
              href="https://en.wikipedia.org/?curid=30863"
              target="_blank"
            >
              Test cricket
            </a>
          </h2>
          <p>
            Test Cricket is the longest format of the sport of cricket, considered the game’s most prestigious and...
          </p>
        </div>
      </div>
    </article>
    <article
      class="m-3 max-w-lg"
    >
      <div
        class="flex flex-row gap-4"
      >
        <div
          class="flex flex-col justify-center"
        >
          <img
            alt="Mantoux test"
            height="33"
            loading="lazy"
            src="https://upload.wikimedia.org/wikipedia/commons/thumb/f/fa/Mantoux_tuberculin_skin_test.jpg/50px-Mantoux_tuberculin_skin_test.jpg"
            width="50"
          />
        </div>
        <div
          class="flex flex-col justify-center"
        >
          <h2>
            <a
              class="text-xl font-bold underline"
              href="https://en.wikipedia.org/?curid=54995"
              target="_blank"
            >
              Mantoux test
            </a>
          </h2>
          <p>
            The Mantoux test or Mendel–Mantoux test (also known as the Mantoux screening test, tuberculin sensitivity...
          </p>
        </div>
      </div>
    </article>
    <article
      class="m-3 max-w-lg"
    >
      <div
        class="flex flex-row gap-4"
      >
        <div
          class="flex flex-col justify-center"
        >
          <img
            alt="Rorschach test"
            height="33"
            loading="lazy"
            src="https://upload.wikimedia.org/wikipedia/commons/thumb/7/70/Rorschach_blot_01.jpg/50px-Rorschach_blot_01.jpg"
            width="50"
          />
        </div>
        <div
          class="flex flex-col justify-center"
        >
          <h2>
            <a
              class="text-xl font-bold underline"
              href="https://en.wikipedia.org/?curid=276250"
              target="_blank"
            >
              Rorschach test
            </a>
          </h2>
          <p>
            The Rorschach test is a projective psychological test in which subjects' perceptions of inkblots are...
          </p>
        </div>
      </div>
    </article>
    <article
      class="m-4 max-w-lg"
    >
      <div
        class="flex flex-col justify-center"
      >
        <h2>
          <a
            class="text-xl font-bold underline"
            href="https://en.wikipedia.org/?curid=519600"
            target="_blank"
          >
            Test match
          </a>
        </h2>
        <p>
          Test match in some sports refers to a sporting contest between national representative teams and may...
        </p>
      </div>
    </article>
    <article
      class="m-4 max-w-lg"
    >
      <div
        class="flex flex-col justify-center"
      >
        <h2>
          <a
            class="text-xl font-bold underline"
            href="https://en.wikipedia.org/?curid=1461217"
            target="_blank"
          >
            Coombs test
          </a>
        </h2>
        <p>
          The direct and indirect Coombs tests, also known as antiglobulin test (AGT), are blood tests used in...
        </p>
      </div>
    </article>
    <article
      class="m-4 max-w-lg"
    >
      <div
        class="flex flex-col justify-center"
      >
        <h2>
          <a
            class="text-xl font-bold underline"
            href="https://en.wikipedia.org/?curid=1858823"
            target="_blank"
          >
            This Is Not a Test!
          </a>
        </h2>
        <p>
          This Is Not a Test! is the fifth studio album by American rapper Missy Elliott, released by The Goldmind...
        </p>
      </div>
    </article>
    <article
      class="m-4 max-w-lg"
    >
      <div
        class="flex flex-col justify-center"
      >
        <h2>
          <a
            class="text-xl font-bold underline"
            href="https://en.wikipedia.org/?curid=2084574"
            target="_blank"
          >
            Test and test-and-set
          </a>
        </h2>
        <p>
          In computer architecture, the test-and-set CPU instruction (or instruction sequence) is designed to implement...
        </p>
      </div>
    </article>
    <article
      class="m-4 max-w-lg"
    >
      <div
        class="flex flex-col justify-center"
      >
        <h2>
          <a
            class="text-xl font-bold underline"
            href="https://en.wikipedia.org/?curid=2249147"
            target="_blank"
          >
            .test
          </a>
        </h2>
        <p>
          .test is a reserved top-level domain used to test websites or web applications as an alternative to testing...
        </p>
      </div>
    </article>
    <article
      class="m-4 max-w-lg"
    >
      <div
        class="flex flex-col justify-center"
      >
        <h2>
          <a
            class="text-xl font-bold underline"
            href="https://en.wikipedia.org/?curid=2691668"
            target="_blank"
          >
            Testability
          </a>
        </h2>
        <p>
          Testability is a primary aspect of science and the scientific method. There are two components to testability...
        </p>
      </div>
    </article>
    <article
      class="m-4 max-w-lg"
    >
      <div
        class="flex flex-col justify-center"
      >
        <h2>
          <a
            class="text-xl font-bold underline"
            href="https://en.wikipedia.org/?curid=3365580"
            target="_blank"
          >
            Patrick's test
          </a>
        </h2>
        <p>
          Patrick's test or FABER test is performed to evaluate pathology of the hip joint or the sacroiliac joint...
        </p>
      </div>
    </article>
    <article
      class="m-4 max-w-lg"
    >
      <div
        class="flex flex-col justify-center"
      >
        <h2>
          <a
            class="text-xl font-bold underline"
            href="https://en.wikipedia.org/?curid=5069303"
            target="_blank"
          >
            This is a test
          </a>
        </h2>
        <p>
          This is a test may refer to:

The testing of the Emergency Alert System
The testing of the former Emergency...
        </p>
      </div>
    </article>
    <article
      class="m-4 max-w-lg"
    >
      <div
        class="flex flex-col justify-center"
      >
        <h2>
          <a
            class="text-xl font-bold underline"
            href="https://en.wikipedia.org/?curid=5200423"
            target="_blank"
          >
            Unterberger test
          </a>
        </h2>
        <p>
          The Unterberger test, also Unterberger's test and Unterberger's stepping test, is a test used in otolaryngology...
        </p>
      </div>
    </article>
    <article
      class="m-3 max-w-lg"
    >
      <div
        class="flex flex-row gap-4"
      >
        <div
          class="flex flex-col justify-center"
        >
          <img
            alt="Thompson test"
            height="40"
            loading="lazy"
            src="https://upload.wikimedia.org/wikipedia/commons/thumb/f/f3/Rupture_tendon_achil%C3%A9en.jpg/50px-Rupture_tendon_achil%C3%A9en.jpg"
            width="50"
          />
        </div>
        <div
          class="flex flex-col justify-center"
        >
          <h2>
            <a
              class="text-xl font-bold underline"
              href="https://en.wikipedia.org/?curid=6883355"
              target="_blank"
            >
              Thompson test
            </a>
          </h2>
          <p>
            The Thompson test (also called Simmonds' test or Simmonds-Thompson test) is used in lower limb examination...
          </p>
        </div>
      </div>
    </article>
    <article
      class="m-4 max-w-lg"
    >
      <div
        class="flex flex-col justify-center"
      >
        <h2>
          <a
            class="text-xl font-bold underline"
            href="https://en.wikipedia.org/?curid=11089416"
            target="_blank"
          >
            Test
          </a>
        </h2>
        <p>
          Test(s), testing, or TEST may refer to:

Test (assessment), an educational assessment intended to measure...
        </p>
      </div>
    </article>
    <article
      class="m-4 max-w-lg"
    >
      <div
        class="flex flex-col justify-center"
      >
        <h2>
          <a
            class="text-xl font-bold underline"
            href="https://en.wikipedia.org/?curid=17447150"
            target="_blank"
          >
            Trendelenburg test
          </a>
        </h2>
        <p>
          The Trendelenburg Test or Brodie–Trendelenburg test  is a test which can be carried out as part of a...
        </p>
      </div>
    </article>
    <article
      class="m-3 max-w-lg"
    >
      <div
        class="flex flex-row gap-4"
      >
        <div
          class="flex flex-col justify-center"
        >
          <img
            alt="Turing test"
            height="38"
            loading="lazy"
            src="https://upload.wikimedia.org/wikipedia/commons/thumb/5/55/Turing_test_diagram.png/50px-Turing_test_diagram.png"
            width="50"
          />
        </div>
        <div
          class="flex flex-col justify-center"
        >
          <h2>
            <a
              class="text-xl font-bold underline"
              href="https://en.wikipedia.org/?curid=21391751"
              target="_blank"
            >
              Turing test
            </a>
          </h2>
          <p>
            The Turing test, originally called the imitation game by Alan Turing in 1950, is a test of a machine...
          </p>
        </div>
      </div>
    </article>
    <article
      class="m-3 max-w-lg"
    >
      <div
        class="flex flex-row gap-4"
      >
        <div
          class="flex flex-col justify-center"
        >
          <img
            alt="TeST Gliders"
            height="38"
            loading="lazy"
            src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/d7/104_b_aircraft_0.jpg/50px-104_b_aircraft_0.jpg"
            width="50"
          />
        </div>
        <div
          class="flex flex-col justify-center"
        >
          <h2>
            <a
              class="text-xl font-bold underline"
              href="https://en.wikipedia.org/?curid=24884283"
              target="_blank"
            >
              TeST Gliders
            </a>
          </h2>
          <p>
            TeST Gliders was a manufacturer of ultralight sailplanes and motorgliders based in Brno, Czech Republic...
          </p>
        </div>
      </div>
    </article>
    <article
      class="m-4 max-w-lg"
    >
      <div
        class="flex flex-col justify-center"
      >
        <h2>
          <a
            class="text-xl font-bold underline"
            href="https://en.wikipedia.org/?curid=28089267"
            target="_blank"
          >
            Litmus test
          </a>
        </h2>
        <p>
          Litmus test may refer to:

Litmus test (chemistry), used to determine the acidity of a chemical solution...
        </p>
      </div>
    </article>
    <article
      class="m-3 max-w-lg"
    >
      <div
        class="flex flex-row gap-4"
      >
        <div
          class="flex flex-col justify-center"
        >
          <img
            alt="Bechdel test"
            height="67"
            loading="lazy"
            src="https://upload.wikimedia.org/wikipedia/commons/thumb/8/86/Alison_Bechdel_at_Politics_and_Prose.jpg/50px-Alison_Bechdel_at_Politics_and_Prose.jpg"
            width="50"
          />
        </div>
        <div
          class="flex flex-col justify-center"
        >
          <h2>
            <a
              class="text-xl font-bold underline"
              href="https://en.wikipedia.org/?curid=32255505"
              target="_blank"
            >
              Bechdel test
            </a>
          </h2>
          <p>
            The Bechdel test ( BEK-dəl), also known as the Bechdel-Wallace test, is a measure of the representation...
          </p>
        </div>
      </div>
    </article>
  </main>
</div>
`;
